sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerAccessor.java

```diff
@@ -229,6 +229,10 @@ static SpannerOptions buildSpannerOptions(SpannerConfig spannerConfig) {
       builder.setCredentials(NoCredentials.getInstance());
     }
     String userAgentString = USER_AGENT_PREFIX + "/" + ReleaseInfo.getReleaseInfo().getVersion();
+    SpannerIOMetadata spannerIOMetadata = SpannerIOMetadata.create();
+    if (spannerIOMetadata.getBeamJobId() != null) {
```

Comment:
there are mixed string == null and string.empty() checks. For robust consider use Strings.isNullOrEmpty in all checks

LGTM after one comment